bsip-0056.md

```diff
@@ -0,0 +1,268 @@
+ownership of the funds and also has limited control over them.
+
+Given that two parties have limited access to the funds, coordination of settlement requests and withdrawal requests
+needs to provide security over the funds. The additional coordination efforts comes with transparency and security to
+both, the customer as well as the 3rd-party centralized service.
+
+From an exchange perspective, a settlement could be seen as synchronizing the user-specific on-chain trading-balances
+with the balances tracked internally. Then, trades happening at the centralized service can be bundled and balances
+settled between trading balances of customers on the chain.
+
+A settlement would then move funds between trading balances of customers. Obviously, each user needs to be able to
+create trading balances that are specific to certain exchange services and settlements can only be applied between
+trading balances for that particular exchange.
+
+This way, the exchange would never hold the funds in custody, but would still be able to allow customers to trade. 
+
+## Trader perspective
+
+A trader can freely deposit money into its exchange-specific trading-balance which can be settled/deposited in the
+exchange either momentarily or after the next settlement performed by the exchange to allow trading with the additional
+funds.
+
+However, withdrawing from the trading balance needs to be limited to ensure pending settlements can properly be executed
+at their time. This means that withdrawal  may require some time to pass (delay). A withdrawal is to be initiated on-chain and the customer funds are being released automatically by the blockchain after the predefined delay.
+This allows the CEX to settle according to their internal database, while still enhancing the security to the 
+user in case of crudities. 
+
+Furthermore consider the case that the CEX is frozen by authorities. Customers can still withdraw their funds, provided that the CEX does not grant access to the settle operation to the authorities right away.
+
+# Specifications
+
+The specifications are separated into the definition and behavioral description of a new kind of vesting balance policy,
+a new operation for settling balances across multiple vesting balances as well as a delay when withdrawing and an operation to deposit funds into an existing vesting balance.
+
+Note:
+ - This new managed vesting balances explicitly counts towards voting power
+
+## Managed Vesting Balance policy
+
+Vesting Balances are *separate* balances that can be created freely by any account on the blockchain. They come with a
+predefined set of policies. Each vesting balance can only use one particular policy  and only contains one 
+balance/asset pair. 
+
+We here propose to add a new vesting policy:
+
+    struct managed_policy_initializer
+    {
+        fc::time_point_sec begin_timestamp;
```

Comment:
No it would not be different, and it also mustn't be part of the API call. It is necessary to calculate when the cliff vesting is over (`begin_timestamp` + `vesting_cliff_seconds`)

Reply:
Then it should not be part of the data contained in the `vesting_balance_create_operation`, but only part of the internal database structure.

Reply:
Done